lib/langchain/conversation.rb

```diff
@@ -84,11 +85,44 @@ def examples
 
     private
 
-    def llm_response(prompt)
-      @llm.chat(messages: @memory.messages, context: @memory.context, examples: @memory.examples, **@options, &@block)
+    def call_llm
+      @llm.chat(messages: transform_messages(messages), context: context&.to_s, examples: transform_messages(examples), **@options, &@block)
     rescue Langchain::Utils::TokenLength::TokenLimitExceeded => exception
       @memory.reduce_messages(exception)
       retry
     end
+
+    def completion_response_path
+      key = @block ? "STREAMING_RESPONSE_PATH" : "COMPLETION_RESPONSE_PATH"
+
+      safe_const_get(llm_klass, key)
+    end
+
+    def transform_messages(messages)
+      role_key = safe_const_get(llm_klass, "ROLE_KEY") || :role
+
+      messages.map do |message|
+        {
+          role_key => role_mapping.fetch(message.type, message.type),
+          :content => message.content
+        }
+      end
+    end
+
+    def role_mapping
+      @role_mapping ||= {
+        "ai" => safe_const_get(llm_klass, "AI_ROLE_KEY"),
+        "human" => safe_const_get(llm_klass, "USER_ROLE_KEY"),
+        "system" => safe_const_get(llm_klass, "SYSTEM_ROLE_KEY")
+      }.compact
+    end
```

Comment:
@andreibondarev I just rebased the PR and fixed the thing about the role getting into the additional_kwargs. This PR is ready to go unless we want to do something with the constants defined in LLMs (`ROLE_KEY`,  `AI_ROLE_KEY`, etc). Do we want to leave it like this or have the Conversation know about the LLMs` response formats?